modules/fundamental/src/purl/service/mod.rs

```diff
@@ -230,41 +231,84 @@ impl PurlService {
         }
     }
 
-    pub async fn purl_by_purl<C: ConnectionTrait>(
+    #[instrument(skip(self, connection), err(level=tracing::Level::INFO))]
+    pub async fn fetch_purl_details<C: ConnectionTrait, I: AsRef<str> + Debug>(
         &self,
-        purl: &Purl,
-        deprecation: Deprecation,
+        identifiers: &[I],
+        deprecated: Deprecation,
         connection: &C,
-    ) -> Result<Option<PurlDetails>, Error> {
-        let canonical = CanonicalPurl::from(purl.clone());
-        match qualified_purl::Entity::find()
-            .filter(qualified_purl::Column::Purl.eq(canonical))
-            .one(connection)
+    ) -> Result<HashMap<String, PurlDetails>, Error> {
+        let (purls, uuids): (Vec<_>, Vec<_>) = identifiers
+            .iter()
+            .partition(|key| key.as_ref().starts_with("pkg:"));
```

Comment:
I didn't love the `partition_map(|either| either)` use before. A closure doing nothing other than returning its argument is a smell.

Reply:
I will read more about loops vs iterators. Thanks